grammar.js

```diff
@@ -1254,7 +1256,7 @@ module.exports = grammar({
       ))
     )),
 
-    element_access_expression: $ => prec.right(PREC.UNARY, seq(
+    element_access_expression: $ => prec.right(PREC.POSTFIX, seq(
```

Comment:
This is fixed to not have conflict in `*p[i]`. The precedence was not matching https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/#operator-precedence.